Anima/filtering/noise_generation/noise_generator/animaNoiseGenerator.cxx

```diff
@@ -0,0 +1,165 @@
+#include <sstream>
+#include <iomanip>
+
+#include <tclap/CmdLine.h>
+
+#include <itksys/SystemTools.hxx>
+
+#include <animaNoiseGeneratorImageFilter.h>
+#include <animaReadWriteFunctions.h>
+
+int main(int argc, char **argv)
+{
+    TCLAP::CmdLine cmd("INRIA / IRISA - VisAGeS Team", ' ',ANIMA_VERSION);
+
+    TCLAP::ValueArg<std::string> inArg("i","inputfile","Input image on which to add noise.",true,"","input image",cmd);
+    TCLAP::ValueArg<std::string> outArg("o","outputfile","Output image. If number of replicates is 1, it is used as provided for saving the noisy dataset with the corresponding file name. If number of replicates > 1, the extension is extracted as desired output file format and the file name without extension is used as base file name for all replicates.",true,"","output file prefix",cmd);
+    TCLAP::ValueArg<std::string> refArg("r","reference","Masked reference image over which the average SNR is defined.",true,"","reference image",cmd);
+    
+    TCLAP::ValueArg<double> snrArg("s","snr","Average SNR in dB over the reference image (default: 25dB).",false,25.0,"mean snr",cmd);
+    TCLAP::ValueArg<unsigned int> repArg("n","num-replicates","Number of independent noisy datasets to generate (default: 1).",false,1,"number of replicates",cmd);
+    
+    TCLAP::SwitchArg gaussArg("G","gauss-noise","Adds Gaussian noise instead of Rician noise.",cmd,false);
+    TCLAP::SwitchArg verboseArg("V","verbose","Outputs noise calculations to the console.",cmd,false);
+
+    TCLAP::ValueArg<unsigned int> nbpArg("p","numberofthreads","Number of threads to run on (default : all cores).",false,itk::MultiThreader::GetGlobalDefaultNumberOfThreads(),"number of threads",cmd);
+
+    try
+    {
+        cmd.parse(argc,argv);
+    }
+    catch (TCLAP::ArgException& e)
+    {
+        std::cerr << "Error: " << e.error() << "for argument " << e.argId() << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    // Put SNR back in linear scale
+    double snr = std::pow(10.0, snrArg.getValue() / 20.0);
+    
+    // Find average B0 value
```

Comment:
En fait, le commentaire devrait être ``// Find average signal value in reference image``. Ce qui donc généralise la chose à autre chose que de la diffusion. Et aussi c'est important car ce qui est parlant pour l'utilisateur lorsqu'on veut bruiter une image, c'est le SNR. Par contre, ce qui est fixe en IRM, c'est la variance du bruit de mesure des bobines. Donc, pour déduire cette variance d'un SNR donné, il faut une valeur de signal référence. D'où mon idée de passer une image de reference et d'en calculer le signal moyen pour s'en servir de signal de référence. Une alternative pourrait être de faire donner le signal moyen directement par l'utilisateur mais ca me semble moins bien. Un avis sur la question ?